### DISCUSSION TITLE
how to 'reset' when server has content

### DISCUSSION THREAD

**Discussion by CarlFK** [videojs/video.js#8644] [General]
setup: usb camera -> gstreamer  libcamerasrc !...! rtmpsink -> nic -> nginx rtmp ... video video.js
this works, when gstreamer is sending a stream.  when it doesn't, the player shows an error.  It shows an error message when there is an error.  good.

What I am trying to do: when it is in this error state and gstreamer starts streaming data, have the player not show the error message and show a play button.

video...player.load() does this.  Except when I click the play button, I get the loading spinner forever, no video.  If I ^r reload the page, then I get a play button, click, see the video stream.  

So, how do I reset the player such that clicking Play plays?

If you want to see what I am describing;
https://ps1.fpgas.mithis.com
turn a pi off and on, the stream will run out of data 60 seconds later and enter the error state.  When the pi boots up and sends "piview: cam" message, the player will show the Play button. click it.  if I haven't fixed it, you will see the spinner.

**Comment by amtins** [comment]
Hi @CarlFK I see that your example uses video.js 7.19.2. A fix has been made in version [8.4.0](https://github.com/videojs/video.js/releases/tag/v8.4.0) which should allow the source to be reloaded when vhs is used. See #8274

Perhaps you could try the latest version of video.js and see if it solves the issue.
